src/main/java/com/akathist/maven/plugins/launch4j/Launch4jMojo.java

```diff
@@ -288,52 +296,61 @@ public void execute() throws MojoExecutionException {
             printState();
         }
 
-        File workdir = setupBuildEnvironment();
-
-        Config c = new Config();
-
-        c.setHeaderType(headerType);
-        c.setOutfile(outfile);
-        c.setJar(getJar());
-        c.setDontWrapJar(dontWrapJar);
-        c.setErrTitle(errTitle);
-        c.setDownloadUrl(downloadUrl);
-        c.setSupportUrl(supportUrl);
-        c.setCmdLine(cmdLine);
-        c.setChdir(chdir);
-        c.setPriority(priority);
-        c.setStayAlive(stayAlive);
-        c.setRestartOnCrash(restartOnCrash);
-        c.setManifest(manifest);
-        c.setIcon(icon);
-        c.setHeaderObjects(relativizeAndCopy(workdir, objs));
-        c.setLibs(relativizeAndCopy(workdir, libs));
-        c.setVariables(vars);
-
-        if (classPath != null) {
-            c.setClassPath(classPath.toL4j(dependencies));
-        }
-        if (jre != null) {
-            c.setJre(jre.toL4j());
-        }
-        if (singleInstance != null) {
-            c.setSingleInstance(singleInstance.toL4j());
-        }
-        if (splash != null) {
-            c.setSplash(splash.toL4j());
-        }
-        if (versionInfo != null) {
-            c.setVersionInfo(versionInfo.toL4j());
-        }
-        if (messages != null) {
-            c.setMessages(messages.toL4j());
+        final File workDir = setupBuildEnvironment();
+        if (infile != null && infile.exists()) {
```

Comment:
it's better to separate these two, if `infile` is not `null` this means that the user defined it, but if `infile` is defined and doesn't exist it means they made a mistake or typo in the file's name so throwing exception here is preferable. With the current approach user will get some strange unrelated warning about missing `<jre/>` or something other.

```java
if (infile != null) {
    if (infile.exists()) {
        try {
            if (getLog().isDebugEnabled()) {
                getLog().debug("Trying to load Launch4j native configuration using file=" + infile.getAbsolutePath());
            }
            ConfigPersister.getInstance().load(infile);
        } catch (ConfigPersisterException e) {
            getLog().error(e);
            throw new MojoExecutionException("Could not load Launch4j native configuration file", e);
        }
    } else {
        throw new MojoExecutionException("Launch4j native configuration file [" + infile.getAbsolutePath() + "] does not exist!", e);
    }
}